File: 1993-sum-of-all-subset-xor-totals/sum-of-all-subset-xor-totals.py
class Solution(object):
    def subsetXORSum(self, nums):
        res, x = 0, len(nums)

        for i in range(1 << x):
            y = 0

            for j in range(x):
                if i >> j & 1:
                    y ^= nums[j]
            
            res += y
        
        return res

File: 1993-sum-of-all-subset-xor-totals/sum-of-all-subset-xor-totals.c
#include <stdio.h>
#include <stdlib.h>

int subsetXORSum(int* nums, int numsSize) {
    int res = 0, x = numsSize;

    for (int i = 0; i < (1 << x); i++) {
        int y = 0;

        for (int j = 0; j < x; j++) {
            if ((i >> j) % 2 == 1) {
                y ^= nums[j];
            }
        }

        res += y;
    }

    return res;
}

